File: Frontend/src/interfaces/v12/stockAdjustment/stockAdjustmentCreateUpdate/stockAdjustmentCreateUpdateDto.ts
import { guid } from "types/guid";

export interface StockAdjustmentCreateUpdateDto {
  number?: string | null;
  warehouseId: guid;
  stockAdjustmentItems: StockAdjustmentItemCreateUpdate[];
}
export interface StockAdjustmentItemCreateUpdate {
  productId: guid;
  quantity: number;
  locationId: guid;
  reason: string;
}

File: Wms.Api/Dto/StockAdjustment/StockAdjustmentCreateUpdate/StockAdjustmentCreateUpdateDto.cs
using Wms.Api.Dto.StockAdjustment.StockAdjustmentCreateUpdate;

namespace Wms.Api.Dto.StockAdjustment.StockAdjustmentCreateUpdateDto
{
    public class StockAdjustmentCreateUpdateDto 
    { 
        public required string Number { get; set; } 
        public Guid WarehouseId { get; set; }
        public ICollection<StockAdjustmentItemCreateUpdateDto>? StockAdjustmentItems { get; set; }
    }
}
